### PR TITLE
Fix CI deploy pipeline

### PR DESCRIPTION
**Pull request by moria97**
## Description
Fix [CI deploy pipeline](https://microsofthealthoss.visualstudio.com/FhirServer/_build?definitionId=29&_a=summary) which is broken due to several issues:
1. Syntax error in referencing add-aad-test-environment.yml job.
2. Wrong config in setting up aad which needs windows platform agent.
3. Wrong schemaAutomaticUpdatesEnabled value in sql deployment tasks.
 
## Related issues
Addresses [issue #].

## Testing
Manual test.

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [x] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [x] Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Skip

### CODE REVIEW

File: build/ci-deploy.yml [hunk 1]
```diff
@@ -114,7 +118,7 @@ stages:
       resourceGroup: $(DeploymentEnvironmentName)
       testEnvironmentUrl: $(TestEnvironmentUrl)
       imageTag: $(ImageTag)
-      schemaAutomaticUpdatesEnabled: 'tool'
+      schemaAutomaticUpdatesEnabled: 'auto'
```
Comment (Ivanidzo4ka) [inline build/ci-deploy.yml:121]: >auto [](start = 38, length = 4)

Can I ask why? I was under impression we always push latest version of schema, and you need to update schema only if your service is living through rollout of new schemas, which shouldn't be the case for CI build.

File: build/ci-deploy.yml [hunk 2]
```diff
@@ -40,7 +40,11 @@ stages:
   - provisionEnvironment
   - DockerBuild
   jobs:
-  - template: ./jobs/add-aad-test-environment.yml
+  - job: setup
+    pool:
+      vmImage: '$(WindowsVmImage)'
```
Comment (Ivanidzo4ka) [inline build/ci-deploy.yml:45]: >WindowsVmImage [](start = 18, length = 14)

Powershell not working on non-windows machines?

**Comment by Ivanidzo4ka** [comment]
Any chance you can add link to example of broken CI pipeline in description?

File: build/ci-deploy.yml (hunk 2 above)
Comment (moria97) [reply build/ci-deploy.yml:45]: It requires higher version to run in other platform. And in our case, it fails to install AzureAD module without this platform selection.

File: build/ci-deploy.yml (hunk 1 above)
Comment (moria97) [reply build/ci-deploy.yml:121]: Yes, I'm also confused here. This pipeline is to replace the deployment in ci resource group. With the parameter set to false, all SQL sites cannot find the stored procedures (seems no initilization) and fail to start.

Comment (Ivanidzo4ka) [reply build/ci-deploy.yml:121]: [https://github.com/microsoft/healthcare-shared-components/issues/196](https://github.com/microsoft/healthcare-shared-components/issues/196)
Seems like a bug to me, I filed an issue.

---
In reply to: [599859457](https://github.com/microsoft/fhir-server/pull/1768#discussion_r599859457) [](ancestors = 599859457)

Comment (LTA-Thinking) [reply build/ci-deploy.yml:121]: For CI setting this to `auto` is correct. We don't want to have to manually update the databases for every change and the CI databases are persistent.

**Comment by LTA-Thinking** [comment]
The script won't run in a PR environment currently. The cleanup script it is referencing had a step added to cleanup docker images, which shouldn't be done in this case as it will prevent the deployments latter in the build from happening. This script needs to copy the resource group deletion part of the cleanup script and run the aad-cleanup job, but not delete docker images.

File: build/ci-deploy.yml (hunk 2 above)
Comment (brendankowitz) [reply build/ci-deploy.yml:45]: This looks ok, I believe the Connect-AAD call in this job required Windows atm

**Comment by moria97** [comment]
> The script won't run in a PR environment currently. The cleanup script it is referencing had a step added to cleanup docker images, which shouldn't be done in this case as it will prevent the deployments latter in the build from happening. This script needs to copy the resource group deletion part of the cleanup script and run the aad-cleanup job, but not delete docker images.

Agree. I will setup another cleanup job

**Comment by moria97** [comment]
> Any chance you can add link to example of broken CI pipeline in description?

Sure, I have updated the description. Thanks

**Review by LTA-Thinking — APPROVED** [review]

**Pull request merged**